File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

map<string, int> m;

string solution(vector<string> participant, vector<string> completion) {
    
    for(string p: participant)
        m[p]++;
    
    for(string p: completion)
        m[p]--;
    
    for(auto tmp : m) {
        if(tmp.second != 0)
            return tmp.first;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> m = new HashMap<>();
        
        for (String p : participant) {
            m.merge(p, 1, Integer::sum);
        }
        
        for (String p : completion) {
            m.merge(p, -1, Integer::sum);
            if (m.get(p) == 0) m.remove(p);
        }
        
        
        return m.keySet().stream().findFirst().get();
    }
}